### PR TITLE
Add markdown any decision record template

### PR DESCRIPTION
**Pull request by lorenyu**
## Ticket
https://github.com/navapbc/platform/issues/3

## Changes
- Add markdown any decision record template
- Add first decision to use MADR

## Context for reviewers
Followed instructions in https://adr.github.io/madr/#initialization

## Testing
None

### CODE REVIEW

**Comment by lorenyu** [comment]
Also created https://github.com/navapbc/template-infra/pull/103, but before I proceed to create them for template-application-nextjs and template-application-flask, I figured I'll review this one first

**Comment by lorenyu** [comment]
@sawyerh also tagging you since I believe you were one of the initial people that pushed for ADRs

File: docs/decisions/0000-use-markdown-architectural-decision-records.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+## Decision Outcome
+
+Chosen option: "MADR 2.1.2", because
+
+* Implicit assumptions should be made explicit.
+  Design documentation is important to enable people understanding the decisions later on.
+  See also [A rational design process: How and why to fake it](https://doi.org/10.1109/TSE.1986.6312940).
+* The MADR format is lean and fits our development style.
+* The MADR structure is comprehensible and facilitates usage & maintenance.
+* The MADR project is vivid.
```
Comment (sawyerh) [inline docs/decisions/0000-use-markdown-architectural-decision-records.md:25]: What a compliment 😂  What's this mean?

File: docs/decisions/index.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Architectural Decision Log
+
+This log lists the architectural decisions for [project name].
```
Comment (sawyerh) [inline docs/decisions/index.md:3]: Should `[project name]` be updated here?

File: docs/decisions/template.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+* [option 1]
+* [option 2]
+* [option 3]
+* … <!-- numbers of options can vary -->
+
+## Decision Outcome
+
+Chosen option: "[option 1]", because [justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].
+
+### Positive Consequences <!-- optional -->
```
Comment (sawyerh) [inline docs/decisions/template.md:30]: Seems redundant with Pros/Cons section but I realize this is a standard MADR thing. Maybe I'm misunderstanding the differences?

**Review by sawyerh — APPROVED** [review]
Excited to have ADRs 🙌🏻

File: docs/decisions/index.md (hunk 2 above)
Comment (lorenyu) [reply docs/decisions/index.md:3]: Hmm good question, I guess for the platform I should put this as `Platform`, but for the other templates like `template-infra` and `template-application-flask` we could have something like `docs/decisions/application` and `docs/decisions/infra/`?

File: docs/decisions/template.md (hunk 3 above)
Comment (lorenyu) [reply docs/decisions/template.md:30]: I think the `Consequences` sections include things like "next steps" and are also intended to allow someone to come back to it later to add what actually happened like a retrospective.

File: docs/decisions/0000-use-markdown-architectural-decision-records.md (hunk 1 above)
Comment (lorenyu) [reply docs/decisions/0000-use-markdown-architectural-decision-records.md:25]: This was all copied over from the madr repo. I think it's a self-compliment by the MADR team :)

**Pull request merged**